modules/table/src/integrationTest/java/org/apache/ignite/internal/disaster/ItDisasterRecoverySystemViewTest.java

```diff
@@ -102,10 +102,40 @@ void testLocalPartitionStatesSystemView() {
         int tableId = getTableId(DEFAULT_SCHEMA_NAME, TABLE_NAME);
 
         assertQuery(localPartitionStatesSystemViewSql())
-                .returns(nodeName0, ZONE_NAME, tableId, DEFAULT_SCHEMA_NAME, TABLE_NAME, 0, HEALTHY.name())
-                .returns(nodeName0, ZONE_NAME, tableId, DEFAULT_SCHEMA_NAME, TABLE_NAME, 1, HEALTHY.name())
-                .returns(nodeName1, ZONE_NAME, tableId, DEFAULT_SCHEMA_NAME, TABLE_NAME, 0, HEALTHY.name())
-                .returns(nodeName1, ZONE_NAME, tableId, DEFAULT_SCHEMA_NAME, TABLE_NAME, 1, HEALTHY.name())
+                .returns(nodeName0, ZONE_NAME, tableId, DEFAULT_SCHEMA_NAME, TABLE_NAME, 0, HEALTHY.name(), 0L)
+                .returns(nodeName0, ZONE_NAME, tableId, DEFAULT_SCHEMA_NAME, TABLE_NAME, 1, HEALTHY.name(), 0L)
+                .returns(nodeName1, ZONE_NAME, tableId, DEFAULT_SCHEMA_NAME, TABLE_NAME, 0, HEALTHY.name(), 0L)
+                .returns(nodeName1, ZONE_NAME, tableId, DEFAULT_SCHEMA_NAME, TABLE_NAME, 1, HEALTHY.name(), 0L)
+                .check();
+    }
+
+    @Test
+    void testLocalPartitionStatesSystemViewWithUpdatedEstimatedSize() throws Exception {
+        assertEquals(2, initialNodes());
+
+        int partitionsCount = 1;
+
+        createZoneAndTable(ZONE_NAME, TABLE_NAME, initialNodes(), partitionsCount);
+        
+        waitLeaderOnAllPartitions(TABLE_NAME, partitionsCount);
+
+        insertPeople(
+                TABLE_NAME,
+                new Person(1, "foo_name", 100.0),
+                new Person(2, "bar_name", 200.0)
+        );
+
+        List<String> nodeNames = CLUSTER.runningNodes().map(Ignite::name).sorted().collect(toList());
+
+        int tableId = getTableId(DEFAULT_SCHEMA_NAME, TABLE_NAME);
+
+        // Small wait is specially added so that the follower can execute the replicated "insert" command and the counter is honestly
+        // increased.
+        Thread.sleep(250);
```

Comment:
Let's just wait till `MvPartitionStorage#estimatedSize()` returns 2 using `waitForCondition()`. Using `sleep()` is unreliable, it will fail from time to time on our TC.